hydra-node/test/Hydra/API/ServerSpec.hs

```diff
@@ -0,0 +1,48 @@
+module Hydra.API.ServerSpec where
+
+import Cardano.Prelude hiding (atomically, threadDelay)
+import Control.Monad.Class.MonadAsync (concurrently_)
+import Control.Monad.Class.MonadSTM (TQueue, atomically, newTQueue, readTQueue, tryReadTQueue, writeTQueue)
+import Control.Monad.Class.MonadTimer (threadDelay)
+import qualified Data.Text as Text
+import Hydra.API.Server (APIServerLog, withAPIServer)
+import Hydra.HeadLogic (ClientResponse (..))
+import Hydra.Ledger.Mock (MockTx)
+import Hydra.Logging (Tracer (..))
+import Network.WebSockets (runClient)
+import Network.WebSockets.Connection (receiveData)
+import Test.Hspec
+import Test.Util (failAfter)
+
+spec :: Spec
+spec = describe "API Server" $ do
+  it "sends response to all connected clients" $ do
+    queue <- atomically newTQueue
+    failAfter 5 $
+      withAPIServer @MockTx "127.0.0.1" 54321 showStdoutTracer noop $ \response -> do
+        threadDelay 0.5
+        withAsync (concurrently_ (testClient queue) (testClient queue)) $ \_ -> do
+          threadDelay 0.5
```

Comment:
Why are the delays necessary here :thinking:  ?

Reply:
Because if the server is not up, the clients will crash, and if the clients are not up, they won't receive the response.